### PR TITLE
Reactor browser component to fulfill dataset 2.0

### PR DESCRIPTION
**Pull request by popcornylu**
**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**
Refactor the browser component. Make it easy to embed in the datasets UI

**Special notes for your reviewer**:

Add search in the shared file page
image

Dataset sample page (This page is hidden from the menu bar, just for demo, it will be implemented in another story sc-22788)
image

**Does this PR introduce a user-facing change?**:
Yes. Add new search bar UI.

```release-note
```

## Test Image
- ~~sc22787-1167c8e~~
- sc22787-8e9c75d

### CODE REVIEW

File: packages/client/src/containers/sharedFiles/__tests__/BrowseSharedFiles.spec.tsx [hunk 1]
```diff
@@ -77,41 +77,35 @@ function setup() {
 
 describe('BrowseSharedFiles', () => {
   it('should render browse shared files with error messages', async () => {
-    const { TestProvider } = setup();
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply packages/client/src/containers/sharedFiles/__tests__/BrowseSharedFiles.spec.tsx:80]: ok

File: packages/client/src/containers/Datasets/DatasetBrowser.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          tabBarExtraContent={tabKey === 'data' ? uploadButton : undefined}
+          onChange={key => setTabKey(key)}
+        >
+          <Tabs.TabPane tab='Data' key='data'>
+            <Browser
+              title={dataset}
+              basePath={`datasets/${dataset}`}
+              path={path || '/'}
+              enabledPHFS={enabledPHFS}
+              onChange={handleChangePath}
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply packages/client/src/containers/Datasets/DatasetBrowser.tsx:77]: No change

File: packages/client/src/components/Browser/Browser.tsx [hunk 3]
```diff
@@ -85,37 +83,34 @@ function humanFileSize(bytes, si = false, dp = 1) {
...
+  changePath,
   onCreate,
-  refetchFiles,
+  style,
 }: {
+  title: string;
   path: string;
   onCreate: () => void;
-  refetchFiles: () => Promise<void>;
+  changePath: (path) => void;
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply packages/client/src/components/Browser/Browser.tsx:110]: Ok

File: packages/client/src/components/Browser/Browser.tsx [hunk 4]
```diff
@@ -282,9 +275,17 @@ interface FileItem {
...
-  path: string;
-  enabledPHFS: boolean;
+export interface BrowserProps {
+  title?: string;
+  basePath?: string;
+  path?: string;
+  enabledPHFS?: boolean;
+  onChange?: (path: string) => void;
+  uploading?: boolean;
+  onUploadingChange?: (boolean) => void;
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply packages/client/src/components/Browser/Browser.tsx:285]: ok

File: packages/client/src/components/Browser/Browser.tsx [hunk 5]
```diff
@@ -124,11 +119,10 @@ function BreadcrumbPaths({
         <Breadcrumb.Item key={i}>
           <a
             onClick={async () => {
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply packages/client/src/components/Browser/Browser.tsx:121]: ok

File: packages/client/src/containers/Datasets/DatasetBrowser.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+              uploading={uploading}
+              onUploadingChange={uploading => {
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply packages/client/src/containers/Datasets/DatasetBrowser.tsx:79]: ok

File: packages/client/src/components/Browser/Browser.tsx [hunk 7]
```diff
@@ -397,31 +412,37 @@ function BrowseSharedFiles({
...
         );
       },
     },
   ];
 
-  function onFilesDeleted({ name }: FileItem) {
-    const basePath = `${appPrefix}g/${groupName}/browse`;
-    const isFolder = name.endsWith('/');
-    const isGroupRootFolder = history.location.pathname === basePath;
+  let dataSource: any[] = get(data, 'files.items', []);
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply packages/client/src/components/Browser/Browser.tsx:435]: ok

File: packages/client/src/components/Browser/Browser.tsx [hunk 8]
```diff
@@ -397,31 +412,37 @@ function BrowseSharedFiles({
...
+  if (!isEmpty(searchText)) {
+    dataSource = dataSource.filter(item => item.name.includes(searchText))
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply packages/client/src/components/Browser/Browser.tsx]: the searchText is from `useState`. The filter will not change the `data.files.items` from props

File: packages/client/src/components/Browser/Browser.tsx [hunk 9]
```diff
@@ -630,4 +640,4 @@ export default compose(
     }),
     name: 'deleteFiles',
   })
-)(BrowseSharedFiles);
+)(Browser) as React.FunctionComponent<BrowserProps>;
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply packages/client/src/components/Browser/Browser.tsx]: ok

**Review by neighborhood999 — APPROVED** [review]

**Pull request merged**